src/reachability/exactReach.jl

```diff
@@ -29,18 +29,19 @@ function solve(solver::ExactReach, problem::Problem)
     return check_inclusion(reach, problem.output)
 end
 
-forward_layer(solver::ExactReach, layer::Layer, input) = forward_layer(solver, layer, convert(HPolytope, input))
+# Forward_linear that coverts the input to HPolytope
+forward_linear(solver::ExactReach, L::Layer, input) = forward_linear(solver, L, convert(HPolytope, input))
 
-function forward_layer(solver::ExactReach, layer::Layer, input::Vector{<:HPolytope})
+# Forward linear for a single polytope and a list of polytopes
+forward_linear(solver::ExactReach, L::Layer, input::HPolytope) = affine_map(L, input)
+forward_linear(solver::ExactReach, L::Layer, input::Vector{<:HPolytope}) = [affine_map(L, set) for set in input]
+
+# Forward act for a single polytope and a list of polytopes
+forward_act(solver::ExactReach, L::Layer, input::HPolytope) = forward_partition(L.activation, input)
+function forward_act(solver::ExactReach, L::Layer, input::Vector{<:HPolytope})
```

Comment:
This is the major downside of splitting `forward_layer` to always be `forward_linear` followed by `forward_act`. We allocate a lot of unnecessary stuff here. Maybe there is something broadcast-y we can do in the definition or call of forward_layer that would help mitigate this...

Reply:
Yeah I'm struggling to come up with a solution, I'm not too familiar with broadcasting.

Reply:
Ok for now we'll live with this double allocation, so I'll resolve it for now